mutt_notmuch.c

```diff
@@ -450,22 +450,25 @@ static struct NmCtxData *get_ctxdata(struct Context *ctx)
   return NULL;
 }
 
-static int string_to_query_type(const char *str)
+static enum NmQueryType string_to_query_type(const char *str)
 {
-  if (!str)
-    str = NmQueryType; /* user's default */
-  if (!str)
-    return NM_QUERY_TYPE_MESGS; /* hardcoded default */
-
-  if (strcmp(str, "threads") == 0)
+  if (mutt_strcmp(str, "threads") == 0)
     return NM_QUERY_TYPE_THREADS;
```

Comment:
@flokli 
I've changed the two `strcmp()`s here to `mutt_strcmp()` which checks for `str` being `NULL`.
(Otherwise mutt would crash if the user unset '$nm_query_type')